src/DropdownBase.js

```diff
@@ -0,0 +1,307 @@
+import React, { cloneElement } from 'react';
+import keycode from 'keycode';
+import DropdownToggle from './DropdownToggle';
+import DropdownMenu from './DropdownMenu';
+import MenuItem from './MenuItem';
+import CustomPropTypes from './utils/CustomPropTypes';
+import createChainedFunction from './utils/createChainedFunction';
+
+import find from 'lodash/collection/find';
+
+const TOGGLE_REF = 'toggle-btn';
+
+export default class DropdownBase extends React.Component {
+  constructor(props, Toggle=DropdownToggle) {
```

Comment:
On second thought, `this.constructor.Toggle` will break everyone using IE <= 10 because static inheritance won't work there, so I'd suggest putting `Toggle` on the prototype, e.g. below do `SplitButton.prototype.Toggle = SplitToggle`.